Bottom: 81e2fda531d9b691d4d1f5927e5d89a4c57de8b6
Top:    3cd650b75c3827c4cb50b12db57205f72d1345dd
Author: Alexey Lyashkov <shadow@Alexeys-MacBook-Pro.local>
Date:   2016-01-21 10:12:50 +0300

attach control message to IB

use IB verbs api to attach control messages
to incomming ring.
Send ctl mgs on wire to verify recv code works



---

diff --git a/ib-sock-ctl.c b/ib-sock-ctl.c
index 4f35dcd..f334383 100644
--- a/ib-sock-ctl.c
+++ b/ib-sock-ctl.c
@@ -42,6 +42,15 @@ ib_sock_ctl_take(struct IB_SOCK *sock)
  */
 int ib_sock_ctl_post(struct IB_SOCK *sock, struct ib_sock_ctl *msg)
 {
+	/* base on ib_mad_post_receive_mads() - that wr isn't need to live any time 
+	 * so - save memory and declare on stack.
+	 */
+	struct ib_recv_wr wr;
+	struct ib_recv_wr *bad_wr;
+	int ret;
+
+	ret = ib_post_recv(sock->is_qp, &wr, &bad_wr);
+
 	return 0;
 }
